Update to API v2

raulg opened the pull request:
Uses both the latest PHP-SDK tool (5.0.1) and the updated API queries.

raulg committed:
Update to API v2

Updates to the app and views to use the latest php sdk and the API v2 queries

raulg committed:
Update image.php

Handle slice types in image_with_caption other than just the default

levimykel commented on app/app.php:
```diff
@@ -17,6 +17,7 @@
 use Prismic\LinkResolver;
 use Prismic\Predicates;
 
+
```
Any reason you added this line?

raulg replied:
None, stray line break

levimykel commented on app/views/bloghome.php:
```diff
@@ -1,38 +1,42 @@
   <div class='blog-avatar' style='background-image: url("<?= $imageUrl ?>");'></div>
-  <h1 class='blog-title'><?= $title ?></h1>
-  <p class='blog-description'><?= $bloghome->getText('blog_home.description') ?></p>
+  <h1 class='blog-title'><?= RichText::asHtml($bloghome->data->headline) ?></h1>
```
You already have the headline title defined above in the $title variable. No need to output this as HTML here.

levimykel commented on app/views/post.php:
```diff
@@ -17,20 +19,19 @@
-    <h1 data-wio-id=<?= $post->getId() ?>>
-      <?= $post->getText('post.title') ? $post->getText('post.title') : "Untitled" ?>
+    <h1 data-wio-id=<?= $post->id ?>>
+      <?= RichText::asHtml($post->data->title, $prismic->linkResolver) ?>
```
You have the title text defined above as $title. No need to output this here as html.

levimykel commented on app/views/slices/firstParagraph.php:
```diff
@@ -1,29 +1,29 @@
-    $firstParagraph = $slice->getPrimary()->getStructuredText('text')->getFirstParagraph();
+foreach ($slices as $slice) {
+  if ($slice->slice_type == "text") {
+    $firstText = RichText::asHtml($slice->primary->text, $linkResolver);
```
Is there a reason you changed this to `$firstText`? I'm fine with that, but just make sure to update the `$firstParagraph` variable above to give this all a default state.

raulg replied:
Total oversight. Changed the variable name to reflect better its nature of no longer being the first paragraph, but forgot about the early empty definition

levimykel commented on app/views/slices/firstParagraph.php:
```diff
@@ -1,29 +1,29 @@
-    $firstParagraph = $slice->getPrimary()->getStructuredText('text')->getFirstParagraph();
+foreach ($slices as $slice) {
+  if ($slice->slice_type == "text") {
+    $firstText = RichText::asHtml($slice->primary->text, $linkResolver);
```
I think you should just get the Rich Text `asText`. In the case where it goes over the limit of 300 characters, this will cut off the closing HTML elements.

raulg replied:
As discussed earlier, while asHtml adds some nice visualization of html formatting in the blog homepage, it introduces the problem of broken tags if the closing is outside the 300 character limit. Falling back to asText for clarity and avoiding first user problems.

levimykel commented on composer.json:
```diff
@@ -1,12 +1,9 @@
 {
   "name": "PHP Quickstart",
```
You should probably update the name of this project while we're at it.

raulg replied:
Changing to "PHP Blog Example", feels like it encapsulates the goal of the project

levimykel commented on app/views/slices/image.php:
```diff
@@ -1,17 +1,18 @@
     echo "<div class='blog-header single' style='background-image: url(" . $imageUrl . ");'><div class='wrapper'>";
     if ( $caption ) {
-      echo "<h1>" . $caption . "</h1>";
+      echo "<h1>" . RichText::asText($caption) . "</h1>";
```
I would probably do the `RichText::asText()` up above when setting the `$caption` variable. This isn't a huge deal, though.

levimykel reviewed (commented):
I haven't done a full test of this yet, but here are my initial comments looking through the changes.

raulg commented on config.php:
```diff
@@ -9,7 +9,7 @@
  * Change this for the URL of your repository.
  */
-define('PRISMIC_URL', 'https://your-repo-name.prismic.io/api');
+define('PRISMIC_URL', 'https://your-repo-name.prismic.io/api/v2');
```
Need to remember to mention this line change in the actual user guide, it's the only required modification in that text to reflect the new code

raulg reviewed (commented):
Comments taken into account, new commit will reflect the requested changes

raulg committed:
Update to V2

Changes for clearer code and avoiding a potential problem when fetching the first text paragraph from blog posts

levimykel reviewed (approved):
Everything looks good, thanks!
